src/settings/__tests__/settings.browser.test.mjs

```diff
@@ -98,6 +98,83 @@ describe('Settings in Browser test suite', () => {
       expect(items).toContainEqual({label, icon});
     });
   });
+  describe('Pane Titles', () => {
+    describe('Services', () => {
+      test('title contains icon', () => {
+        // Services pane is shown by default
+        const title = document.querySelector('h2.title');
+        expect(title).not.toBeNull();
+        const iconElement = title.querySelector('.material3.icon');
+        expect(iconElement).not.toBeNull();
+        expect(iconElement.textContent).toBe('\ue5c3');
+      });
+
+      test('title contains text', () => {
+        // Services pane is shown by default
+        const title = document.querySelector('h2.title');
+        expect(title).not.toBeNull();
+        expect(title.textContent).toContain('Services');
+      });
+    });
+
+    describe('Spell check', () => {
+      let spellCheckButton;
+
+      beforeEach(async () => {
+        spellCheckButton = Array.from(document.querySelectorAll('.navigation-rail-button'))
+          .find(button => button.querySelector('.navigation-rail-button__label').textContent === 'Spell check');
+        fireEvent.click(spellCheckButton);
+        await waitFor(() => document.querySelector('h2.title'));
+      });
+
+      test('title contains icon', async () => {
+        await waitFor(() => {
+          const title = document.querySelector('h2.title');
+          expect(title).not.toBeNull();
+          const iconElement = title.querySelector('.material3.icon');
+          expect(iconElement).not.toBeNull();
+          expect(iconElement.textContent).toBe('\ue8ce');
+        });
+      });
+
+      test('title contains text', async () => {
+        await waitFor(() => {
+          const title = document.querySelector('h2.title');
+          expect(title).not.toBeNull();
+          expect(title.textContent).toContain('Spell check');
+        });
+      });
+    });
+
+    describe('Other', () => {
+      let otherButton;
+
+      beforeEach(async () => {
+        otherButton = Array.from(document.querySelectorAll('.navigation-rail-button'))
+          .find(button => button.querySelector('.navigation-rail-button__label').textContent === 'Other');
+        fireEvent.click(otherButton);
+        await waitFor(() => document.querySelector('h2.title'));
+      });
+
+      test('title contains icon', async () => {
+        await waitFor(() => {
+          const title = document.querySelector('h2.title');
+          expect(title).not.toBeNull();
+          const iconElement = title.querySelector('.material3.icon');
```

Comment:
Why aren't you just using the querySelector to retrieve the icon component directly (`h2.title .material3.icon`)
The title content is asserted in a different test case.